### PR TITLE
crypto_util: Refactor, improve test coverage [WIP]

### PR DESCRIPTION
**Pull request by ArchimedesPi**

### CODE REVIEW

**Comment by ArchimedesPi** [comment]
`_pyopenssl_load` can be merged with `pyopenssl_load_certificate`.

File: letsencrypt/tests/crypto_util_test.py [hunk 1]
```diff
@@ -212,6 +212,17 @@ def test_parse_no_sans(self):
...
+    def test_pyopenssl_load(self):
+        from letsencrypt.crypto_util import _pyopenssl_load
+
+        method_mock = mock.MagicMock()
+        fake_data = "this is test data"
+        fake_types = ('fake loading type',)
+        _pyopenssl_load(fake_data, method_mock, fake_types)
+        method_mock.assert_called_with(fake_types[0], fake_data)
+
+
```
Comment (kuba) [inline letsencrypt/tests/crypto_util_test.py]: too much whitespace here, 2 lines should be used for separation

File: letsencrypt/tests/crypto_util_test.py [hunk 2]
```diff
@@ -212,6 +212,17 @@ def test_parse_no_sans(self):
         self.assertEqual(
             [], self._call(test_util.load_vector('csr-nosans.pem')))
 
+class PyOpenSslLoaderTest(unittest.TestCase):
```
Comment (kuba) [inline letsencrypt/tests/crypto_util_test.py]: nit1: not enough whitespace above that line
nit2: `PyOpenSSLLoadTest`
nit3: missing docstring: `"""Tests for letsencrypt.crypto_util._pyopenssl_load"""`
nit4; missing whitespace between `class` and `def`

hey, you may want to run `pep8` before submitting stuff :)

File: letsencrypt/tests/crypto_util_test.py [hunk 3]
```diff
@@ -212,6 +212,17 @@ def test_parse_no_sans(self):
...
+    def test_pyopenssl_load(self):
+        from letsencrypt.crypto_util import _pyopenssl_load
+
+        method_mock = mock.MagicMock()
+        fake_data = "this is test data"
+        fake_types = ('fake loading type',)
```
Comment (kuba) [inline letsencrypt/tests/crypto_util_test.py]: nit: "Be consistent with your choice of string quote character within a file." as per https://google-styleguide.googlecode.com/svn/trunk/pyguide.html?showone=Strings#Strings.

**Comment by kuba** [comment]
Hah, everything mocked out; maybe we should actually try to run this on real data, without changing `method`?

Put down all the nits I could spot, so you know what to watch out in later contributions: in general `pep8` and `pylint` are you friends. Try to match the existing style. :)

**Comment by pde** [comment]
LGTM

**Pull request merged**